### ISSUE TITLE
Process substition in BASH fails.

### ISSUE THREAD

**Issue by kherge** [tbodt/ish#164]
## Example Script

```sh
#!/usr/bin/env bash

while read LINE; do
    echo “$LINE”
done < <(echo hello)
```

## Expected Result

    $ hello

## Actual Result

    line 5: /dev/fd/62: No such file or directory

## References

1. http://www.gnu.org/software/bash/manual/html_node/Process-Substitution.html#Process-Substitution
2. https://stackoverflow.com/a/2443120

**Comment by tbodt** [comment]
What's missing now is the very weird (to me) linux behavior where pipe file descriptors appear to be broken symlinks but can actually be opened, and become a second file descriptor for the same pipe.

**Comment by xinity** [comment]
@tbodt any hints to fix this ? as an example we can't leverage powerline10k or any advanced (and very useful features) of ZSH :(

Do you need more tests from us ? 
let me know :)

Keep up the awesome work, discovered your app few days ago , NIFTY !!!!

**Comment by tbodt** [comment]
The problem is well understood (https://github.com/ish-app/ish/issues/164#issuecomment-459837509) but I am not quite sure how to solve it.

**Comment by f-eric** [comment]
This is pretty much over my head but I've been looking around a bit and I stumble upon https://github.com/microsoft/WSL/issues/266 which addresses the same thing(?). I'm not sure if it's helpful, but maybe...

**Comment by tbodt** [comment]
Yeah, WSL had the same problem. Not much info there about how they solved it though.

**Comment by kherge** [comment]
I apologize that this is also over my head, but maybe it's buried in one or more commits for their kernel? I tried to create [a narrow search](https://github.com/search?q=proc+fd+repo%3Amicrosoft%2FWSL2-Linux-Kernel+created%3A%3E2016-04-25+created%3A%3C2016-06-08&type=Commits&ref=advsearch&l=&l=) that could hopefully be a starting point.

**Comment by tbodt** [comment]
The bug report you linked is for WSL 1, which is not open source. This was never a problem in WSL 2.

**Comment by f-eric** [comment]
I wish I was a developer... For what it's worth I asked in the WSL issue if there would be any possibility for them to share how it was solved. I'm not sure how I'm able to help in any other way. The worst that can happen is no one answers or we get a no.

**Comment by makakin** [comment]
Any solution to fix this issue?

**Comment by le0-VV** [comment]
Is this thread still going? Came across this problem and is now mildly infuriating me

**Comment by kamikaze3384** [comment]
j`aimerais que mon compte sois supprimé

**Comment by saagarjha** [comment]
Hi @kamikaze3384! If you're looking to delete your GitHub account, you should go through [their steps to do so](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-user-account/managing-user-account-settings/deleting-your-user-account). This unfortunately is not the right place for those requests :(

**Comment by ryleu** [comment]
@tbodt there seems to be a solution now, so I'd like to bring this to your attention again.

**Comment by annpocoyo** [comment]
What’s weird is that the pipe still gets created.
As seen by running `ls /proc/self/fd <(echo “test”)`, even if you make a symlink to /proc/self/fd from /dev/fd as seen by running `ls /dev/fd <(echo “test”)`. Someone please merge the pull request

**Comment by kalebmckale** [comment]
@annpocoyo: I'm seeing this on mine:
```sh
  File: '/dev/fd/63' -> 'anon_inode:[unknown]'
  Size: 0               Blocks: 0          IO Block: 0      symbolic link
Device: 0h/0d   Inode: 17732923551252482  Links: 0
Access: (0777/lrwxrwxrwx)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 1970-01-01 00:00:00.000000000 +0000
Modify: 1970-01-01 00:00:00.000000000 +0000
Change: 1970-01-01 00:00:00.000000000 +0000
```
when I run stat. The symlink is indeed created, but it's not linked to a pipe as it should be (from what I gather from a bunch of reading I've been doing).

More so, I noticed strangely that running processes in general are not being created as one would expect:

```sh
> sleep 1000 &
[1] 221
> lsof -p 221
COMMAND PID USER   FD      TYPE DEVICE   SIZE    NODE NAME
sleep   221 root  cwd   unknown                       /proc/221/cwd (readlink: No such file or directory)
sleep   221 root  rtd   unknown                       /proc/221/root (readlink: No such file or directory)
sleep   221 root  txt       REG    1,8 808104 4317863 /bin/busybox
sleep   221 root    0u      CHR  136,0              3 /dev/pts/0
sleep   221 root    1u      CHR  136,0              3 /dev/pts/0
sleep   221 root    2u      CHR  136,0              3 /dev/pts/0
```

I'm not sure this is exactly related but since I discovered this during my reading and research, I figured it might be useful?

**Comment by rrr9rrr** [comment]
My testing results:

iPhone7nABEL:~# ls -la <(:) /proc/self/fd
lrwxrwxrwx 0 root root 0 Jan  1  1970 /dev/fd/63 -> 'anon_inode:[unknown]'

/proc/self/fd:
total 0
lrwxrwxrwx 0 root root 0 Jan  1  1970 0 -> /dev/pts/0
lrwxrwxrwx 0 root root 0 Jan  1  1970 1 -> /dev/pts/0
lrwxrwxrwx 0 root root 0 Jan  1  1970 2 -> /dev/pts/0
lrwxrwxrwx 0 root root 0 Jan  1  1970 3 -> /proc/35/fd
lrwxrwxrwx 0 root root 0 Jan  1  1970 63 -> 'anon_inode:[unknown]'
iPhone7nABEL:~# ls -la <(:) /dev/fd
lrwxrwxrwx 1 root root 13 Nov 29 07:41 /dev/fd -> /proc/self/fd
lrwxrwxrwx 0 root root  0 Jan  1  1970 /dev/fd/63 -> 'anon_inode:[unknown]'
iPhone7nABEL:~# 

iSH 1.3.2 (Build 494)
![A66EDE32-51A8-4043-9EC9-DE96DAE58FC7](https://github.com/user-attachments/assets/a3fd3bf4-f2f3-4576-9cb8-d3cba0826537)
![540F69ED-EB2C-48AA-8F45-05E2B93FDD68](https://github.com/user-attachments/assets/bd6fbf77-b291-4879-89dc-fb9ab16ce6bb)
